lib/galaxy/tools/parameters/validation.py

```diff
@@ -133,7 +133,7 @@ def from_element(cls, param, elem):
 
     def __init__(self, message, expression, negate):
         if message is None:
-            message = f"Value '%s' does {'not ' if negate == 'false' else ''}match regular expression '{expression}'"
+            message = f"Value '%s' does {'not ' if negate == 'false' else ''}match regular expression '{expression.replace('%', '%%')}'"
```

Comment:
What is the problem solved here?

Reply:
It's in the commit message:
```
In [1]: expression = "5 %"
   ...: negate = False
   ...: value = 1
   ...: f"Value '%s' does {'not ' if negate == 'false' else ''}match regular expression '{expression}'" % value
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-1754553231d4> in <module>
      2 negate = False
      3 value = 1
----> 4 f"Value '%s' does {'not ' if negate == 'false' else ''}match regular expression '{expression}'" % value

TypeError: not enough arguments for format string
```
With the fix:
```
In [2]: f"Value '%s' does {'not ' if negate == 'false' else ''}match regular expression '{expression.replace('%', '%%')}'" % value
Out[2]: "Value '1' does match regular expression '5 %'"

```